straxen/plugins/nveto_pulse_processing.py

```diff
@@ -84,12 +85,35 @@ def clean_up_empty_records(records, only_last=True):
     for ind, r in enumerate(records):
         if only_last:
             m_last_fragment = (r['record_i'] > 0) and (r['length'] < len(r['data']))
-            if m_last_fragment:
+            if not m_last_fragment:
                 indicies_to_keep[n_indicies] = ind
                 n_indicies += 1
                 continue
 
         if np.any(r['data'] != 0):
             indicies_to_keep[n_indicies] = ind
             n_indicies += 1
+            continue
+
+        # If we arrive here this means we will remove this record
+        # Hence we have to update the pulse_lengths accordingly:
+        length = r['length']
+
+        TRIAL_COUNTER = 500  # Do not want to be stuck forever
+        for neighbors in record_links:
```

Comment:
can you enumerate over left and right, and if only_last, you can skip right and save time. 

Also, I was first confused first but see now it makes sense but making left right explicit makes it easier to understand.